github/resource_github_repository_ruleset_utils.go

```diff
@@ -0,0 +1,256 @@
+package github
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-github/v53/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func buildRulesetRequest(d *schema.ResourceData, sourceType *string) (*github.Ruleset, error) {
+	target := d.Get("target").(string)
+	req := &github.Ruleset{
+		Name:        d.Get("name").(string),
+		Target:      &target,
+		SourceType:  sourceType,
+		Enforcement: d.Get("enforcement").(string),
+	}
+
+	bypassActors, err := expandBypassActors(d)
+	if err != nil {
+		return nil, err
+	}
+	req.BypassActors = bypassActors
+
+	conditions, err := expandConditions(d)
+	if err != nil {
+		return nil, err
+	}
+	rulesetConditions := github.RulesetConditions{
+		RefName:        conditions,
+		// RepositoryName: &github.RulesetRepositoryConditionParameters{}, // TODO: Implement for org stuff
```

Comment:
It shouldn't be that hard to build on this to also implement Org level Rules. Leaving this until the schema has been finalized and `go-github` has been bumped

Reply:
Or another PR, idk what makes sense. We'll see